new/analyse/widen-dependency-ranges.ts

```diff
@@ -0,0 +1,32 @@
+import type { Module } from '../types';
+
+import { clone } from 'lodash';
+
+import { logger } from '../helpers/logger';
+
+const { log } = logger('[analyse] wide dependnecy ranges');
+
+// Matches either "^", "~" or "*" at the beginning of the string
+const LOOSE_RANGE_PATTERN = /^[~^*]/;
+
+export const widenDependencyRanges: Module = (env) => {
+  const packagesClone = clone(env.packages);
+
+  log('Analysing packages');
+  let widenCounter = 0;
+  Object.values(packagesClone!).forEach((packageValue) => {
+    // Widen dependencies ranges
+    packageValue.dependsOn.forEach((dependency) => {
+      // except they're already widened
+      if (LOOSE_RANGE_PATTERN.test(dependency.range)) {
```

Comment:
I'd really suggest to use the semver package here. There are much more loose range patterns like `2 - 3` or `>=4 <6 ... Another possibility is to test for the other way round (`<number>.<number>.<number>`).